torch/_higher_order_ops/associative_scan.py

```diff
@@ -63,8 +92,15 @@ def associative_scan(
             This function must be pure, pointwise, and satisfy the associative property.
         input (torch.Tensor): The input tensor, or nested pytree of tensors.
             All inputs are expected to have the same shape.
-        dim (int): the dimension to scan over
-
+        dim (int): The dimension to scan over
+        reverse (bool): A boolean stating if the scan should be reversed with respect to the dimension.
+        generic_scan (bool): A boolean stating whether a generic scan mode should be used. 
+            If the generic scan mode is ``not used``, there are restrictions on the operations allowed
+            within the ``combine_fn``. For example, only pointwise functions are currently allowed.
+            If the generic scan mode is ``used``, the restrictions on the combine_fn are less strict, 
+            for example also non-pointwise operations are allowed. 
+            However, the non-generic scan mode utilizes efficient ``tl.associative_scan`` calls and is thus
+            more efficient as the generic scan mode.
 
     Example::
```

Comment:
Can you add an example of `generic_scan=True` (one with matmul would be good) and one with two inputs and outputs (e.g. something like `cummin`).
In general, I would like to show that the API for `generic_scan=True` and `=False` is different (!!) when there are multiple inputs and outputs.

Reply:
I tried to unify the API and hence this may not be necessary anymore?